### ISSUE TITLE
Changes are not written to file

### ISSUE THREAD

**Issue by wgslr** [mpcjanssen/simpletask-android#741]

**Comment by mpcjanssen** [comment]
This is an android issue. You can't  (easily) have write permission to the sdcard. I should probably show a message if you open a read only file but besides that there is not much that can be done.
There are ways to save files to extsd but file monitoring is not possible.

**Comment by mpcjanssen** [comment]
I will add a message if writing the file fails.

**Comment by wgslr** [comment]
Thank you for quick response.
I was just testing things, so I didn't lose anything important. Some apps show this dialog where you have to select sdcard and grant permissions, but I guess it wouldn't allow file monitoring? Then error message will have to suffice.

**Comment by mpcjanssen** [comment]
@wgslr exactly. Showing the dialog also requires a lot of code changes to make it work. I looked into it when MM was released.

**Comment by mpcjanssen** [comment]
Also (and this is blocking) the access needed for external SD cards will not allow monitoring of the file for changes.

**Comment by smichel17** [comment]
We can't watch an external file but we could poll for changes. Although, do we want to open that can of worms?

Perhaps when you select a file on an external card we display a dialog to the effect of "External SD cards are not supported, do you want to (1) Copy this file to the internal card, (2) Import the contents to your current todo.txt file, or (3) cancel?"

**Comment by mpcjanssen** [comment]
@smichel17 all this is just a PITA and I indeed don't want to open that can of worms. External SDCard devices are dying out anyway (unfortunately)

**Comment by mlncn** [comment]
And i take it that intermittently syncing from Simpletask storage to the SDcard 1) is not trivial and 2) shouldn't be part of Simpletask anyway?  I can't find an application that would do this in F-Droid— if anyone has any pointers.  (Use case of having a todo.txt available on my computer via a synced ownCloud/nextCloud folder.  The sync doesn't have to be clever, as i'll only ever be editing in one place at a time.)

Also, just for clarification, does the "Root directory" under "Other configuration" therefore do anything at all?

Thanks so much for this great app and your continued work on it.  It's the best hope i have for getting organized and getting things done!

**Comment by mpcjanssen** [comment]
@mlncn maybe something like foldersync can do local sync (from internal storage to sdcard)
